Action copied from smartly-ui

mhalmagiu opened the pull request:
Copied this functionality from our UI repo

Commit:
Action copied from smartly-ui

ramert commented on index.js:
```diff
@@ -2,14 +2,41 @@ const core = require('@actions/core');
+  const minorLabel = label_array.includes('minor');
+  const patchLabel = label_array.includes('patch');
+
+  if (majorLabel === true && minorLabel === false && patchLabel === false) {
```
Includes always return boolean so this could be written,
`if (majorLabel && !minorLabel && !patchLabel) {`

Still feels a bit complex to do the comparison like this, but...

ramert commented on index.js:
```diff
@@ -2,14 +2,41 @@ const core = require('@actions/core');
+  VERSION_LOWER = VERSION_UPPER.toLowerCase();
+  // Return values for use in next step
+  core.setOutput('VERSION_UPPER', VERSION_UPPER);
+  core.setOutput('VERSION_LOWER', VERSION_LOWER);
```
VERSION_LOWER is not reassigned, so you could just have here VERSION_UPPER.toLowerCase();

ramert commented on index.js:
```diff
@@ -2,14 +2,41 @@ const core = require('@actions/core');
+  const enforceSet = core.getInput('enforce');
+
+  // Predefining default outputs
+  let VERSION_UPPER = '';
```
Typical naming convention in JS is camelCase "always".

Commit:
Refactor index.js to use functions

Instead of using comments to describe what the code does, refactor them
inside functions describing the functionality.

This way the code is more modular and easier to develop further.

Also started to use js 'enum' to reference to different versions.

Commit:
camel case function names

Commit:
Minor refactoring for version checking

mhalmagiu committed:
Merge pull request #3 from smartlyio/refactor-index-js

Refactor index.js to use functions
